Core/Object Arts/Dolphin/Base/Package.cls

```diff
@@ -42,6 +42,15 @@ aboutBlockBytes: aByteArray
 
 !
 
+addAllPrerequisitesTo: aSet
+	"Private - Add to <aSet> all those Packages which must be loaded in
+	before the receiver may be loaded in."
+
+	self prerequisites
+		do: [:each | (aSet includes: each) ifFalse:
+					[aSet add: each.
```

Comment:
Double lookup (once to check inclusion and once to add) can be avoided here by using Set>>addNewElement:. This adds the element if not already present returning true if it was added, or false if already in the set.